dev-docs/hail-for-new-engineers.md

```diff
@@ -0,0 +1,189 @@
+# Hail for New Engineers
+
+This document introduces the entire Hail project to software engineers. We assume no previous
+knowledge of Hail.
+
+## A Brief History
+
+The Hail project started in 2015 to facilitate the interactive analysis of large human genetic
+datasets. Hail was a command-line program that used Apache Spark to run analysis on partitioned
+genetic datasets simultaneously using hundreds of computer cores. To use Hail, a user needs an
+Apache Spark cluster. Most Hail users use Google Dataproc Spark clusters.
+
+The essential feature of a human genetic dataset is a two-dimensional matrix of human
+genotypes. Every genotype has the property "number of alternate alleles". This property allows a
+matrix of genotypes to be represented as a numeric matrix. Geneticists use linear algebraic
+techniques on this numeric matrix to understand the relationship between human disease and human
+genetics.
+
+In November of 2016, the Hail command-line interface was eliminated and a Python interface was
+introduced. During this time, Hail was not versioned. Users had to build and use Hail directly from
+the source code repository.
+
+In March of 2017, Hail team began work on a compiler.
+
+In October of 2018, the Hail Python interface was modified, in a backwards-incompatible way. This
+new Python interface was named "Hail 0.2". The old Python interface was retroactively named "Hail
+0.1". Hail team began deploying a Python package named `hail` to [PyPI](https://pypi.org). The Hail
+python package version compiles with [Semantic Versioning](https://semver.org).
+
+Meanwhile, in September of 2018, Hail team began work on a system called "Batch". Batch runs
+programs in parallel on a cluster of virtual machines. Also in September, Hail team began work on a
+system called "CI". CI is an acronym for Continuous Integration. CI runs the tests for every pull
```

Comment:
"CI (continuous integration)" rather than "CI is an acronym for Continuous Integration."